packages/swc-configs/tests/build.test.ts

```diff
@@ -0,0 +1,32 @@
+import { DefaultBuildConfig, defineBuildConfig } from "../src/build.ts";
+import type { ConfigOverride } from "../src/resolveOverrides.ts";
+
+test("when no options are provided, return the default config", () => {
+    const result = defineBuildConfig();
+
+    expect(result).toEqual(DefaultBuildConfig);
+});
+
+test("when \"parser\" is \"ecmascript\", the configuration parser is ecmascript", () => {
+    const result = defineBuildConfig({
+        parser: "ecmascript"
+    });
+
+    expect(result).toMatchSnapshot();
+});
+
+test("when a config override function is provided, the function argument is the config with the non config override options applied", () => {
+    const expectedArgument = defineBuildConfig({
+        parser: "ecmascript"
+    });
+
+    const fct = jest.fn(() => ({
```

Comment:
`jest.fn<Config, [Config]>()` would type your function better, and you wouldn't have to cas it on line 28

Reply:
Nice one, thanks